views/pages/NewConfigPage.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useMemo, useState, useEffect } from 'react'
+import { useForm } from 'react-hook-form'
+import { useOutletContext } from 'react-router-dom'
+import Snackbar from '@material-ui/core/Snackbar'
+
+import api from '../api'
+import useArrayFormFields from '../hooks/useArrayFormFields'
+import ConfigForm from '../forms/ConfigForm'
+import { UserConfigModel } from '../models'
+
+const NewConfigPage = () => {
+  const { user, classes } = useOutletContext()
+  const { uid } = user
+  const [gridState, setGridState] = useState({ columns: null, gridWidth: 300 })
+  const [snackBar, setSnackBar] = useState({
+    open: false,
+    message: '',
+  })
+  const defaultValues = UserConfigModel.defaultFormValues({
+    readers: [{ value: uid, label: uid }],
+  })
+  const defaultFieldValue = UserConfigModel.defaultConfigValues
+  const { handleSubmit, control } = useForm({ defaultValues })
+  const { fields, addNewField, removeField } = useArrayFormFields({
+    control,
+    name: 'config',
+    defaultFieldValue,
+  })
+  const colors = useMemo(() => UserConfigModel.configColorOptions(), [])
+  const friendsList = useMemo(
+    () => user.friends.map((friend) => ({ label: friend, value: friend })),
+    [user]
+  )
+  const handleFormData = async (formValues) => {
+    try {
+      const newConfigurationAttributes = await UserConfigModel.processNewConfig(
+        formValues,
+        colors,
+        uid
+      )
+
+      await api.userConfigurations.create(uid, newConfigurationAttributes)
+
+      setSnackBar({ open: true, message: UserConfigModel.successMessage })
+    } catch (error) {
+      setSnackBar({ open: true, message: error.message })
+    }
+  }
+  const handleResize = () => {
+    if (window.innerWidth >= 768) {
+      const columns = Math.floor(window.innerWidth / gridState.gridWidth)
+      const gridWidth = window.innerWidth / columns
+      setGridState({ columns, gridWidth })
+    } else if (gridState.columns !== 1) {
+      const gridWidth = window.innerWidth / 1
```

Comment:
Doesn't anything divided by 1 return itself?

Reply:
Good catch. I updated how we calculate the grid state since this logic is being used elsewhere it is now a hook that will be imported. [df8f7a1](https://github.com/AMP-SCZ/dpdash/pull/332/commits/df8f7a1d6516bbba1d9fbf2bfad2ca3e476557d1)